src/irrlichttypes.h

```diff
@@ -20,47 +20,34 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #ifndef IRRLICHTTYPES_HEADER
 #define IRRLICHTTYPES_HEADER
 
-/* Ensure that <stdint.h> is included before <irrTypes.h>, unless building on
- * MSVC, to address an irrlicht issue: https://sourceforge.net/p/irrlicht/bugs/433/
- *
- * TODO: Decide whether or not we support non-compliant C++ compilers like old
- *       versions of MSCV.  If we do not then <stdint.h> can always be included
- *       regardless of the compiler.
- */
-#ifndef _MSC_VER
-#	include <stdint.h>
-#endif
+namespace irr {
+	class IrrlichtDevice;
+	struct SEvent;
 
-#include <irrTypes.h>
+	namespace video {};
+	namespace scene {};
+	namespace core {};
+	namespace gui {};
+	namespace io {};
+};
 
-using namespace irr;
+using irr::IrrlichtDevice;
+using irr::SEvent;
 
-// Irrlicht 1.8+ defines 64bit unsigned symbol in irrTypes.h
-#if (IRRLICHT_VERSION_MAJOR == 1 && IRRLICHT_VERSION_MINOR < 8)
-#ifdef _MSC_VER
-	// Windows
-	typedef long long s64;
-	typedef unsigned long long u64;
-#else
-	// Posix
-	typedef int64_t s64;
-	typedef uint64_t u64;
-#endif
-#endif
+namespace video = irr::video;
+namespace scene = irr::scene;
+namespace core = irr::core;
+namespace gui = irr::gui;
+namespace io = irr::io;
 
-#define S8_MIN  (-0x7F - 1)
```

Comment:
Its also more consistent with the actual types. int8 isn't a valid type in minetest code, but s8 is.